components/usage/pkg/stripe/stripe.go

```diff
@@ -122,6 +137,47 @@ func (c *Client) updateUsageForCustomer(ctx context.Context, customer *stripe.Cu
 	}, nil
 }
 
+// GetUpcomingInvoice fetches the upcoming invoice for the given team or user id.
+func (c *Client) GetUpcomingInvoice(ctx context.Context, in *v1.GetUpcomingInvoiceRequest) (*StripeInvoice, error) {
```

Comment:
The stripe client in this package shouldn't be taking on a dependency from the API. It's tightly coupling the Stripe client to the API when they should be each able to evolve independently. 

In this case, I'd supply either teamID or userID as a string to break this coupling.